File: FrontEnd/src/app/models/section.ts
import { GetCourseDto } from "./course";
import { TeacherDto } from "./teacher";

export class SectionForAddUodate {
  id: number;
  name: string;
  courseId: number;
  teacherId: number;
  totalHoures: number;
  lecturesPerWeek: number;
  timeFrom: number;
  timeTo: number;

}

export class SectionData {
  id: number;
  name: string;
  totalHoures: number;
  lecturesPerWeek: number;
  timeFrom: number;
  timeTo: number;
  teacher: TeacherDto;
  course: GetCourseDto;  //course.department
}

File: API/Models/Domain/Section.cs
namespace API.Models.Domain
{
    public class Section
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public int TotalHoures { get; set; }
        public int LecturesPerWeek { get; set; }
        public int? TimeFrom { get; set; }
        public int? TimeTo { get; set; }
        public virtual Course? Course { get; set; } 
        public virtual int CourseId { get; set; }
        public virtual User? Teacher { get; set; }
        public virtual int TeacherId { get; set; }
    }
}
